### 1
#### Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [18]:
from math import factorial
from numpy import exp

In [40]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n-k)))
def bernulli_probability(n, k, p):
    return combinations(n, k) * p**k * (1-p)**(n-k)
def puasson_probability(n, m, p):
    return (n*p)**m/factorial(m)*e**(-n*p)

In [41]:
bernulli_probability(100, 85, 0.8)

0.048061793700746355

### Ответ:
$P(A) = 0.048 \approx 5\%$

### 2
#### Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день?

In [42]:
puasson_probability(5000, 0, 0.0004)

0.1353352832366127

### Ответ:
$P(A) = 0.135 \approx 14\%$

### 3
#### Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [43]:
bernulli_probability(144, 70, 0.5)

0.06281178035144776

### Ответ:
$P(A) = 0.062 \approx 6\%$

### 4
#### В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

a.

Вероятность, что из обоих ящиков достанут по 2 белых шара равна:

$\frac{C_{7}^2 * C_{9}^2}{C_{10}^2 * C_{11}^2} \approx 0.3 = 30\%$

Если считать по формуле Бернулли, получаем ответ $0.33\%$

In [50]:
(combinations(7,2)*combinations(9,2))/(combinations(10,2)*combinations(11,2))

0.3054545454545455

In [74]:
bernulli_probability(2, 2, 9/11) * bernulli_probability(2, 2, 7/10)

0.32801652892561983

b.

Вероятность, что вытащат ровно 2 белых шара равна вероятности того, что из обоих вытащат по одному, или из первого 2 и из второго 0, или из первого 0 и из второго 2

$P(A) = 0.2 = 20\%$

In [73]:
(bernulli_probability(2, 1, 9/11) * bernulli_probability(2, 1, 7/10)) + \
(bernulli_probability(2,0,9/11) * bernulli_probability(2,2,7/10))+ \
(bernulli_probability(2,2,9/11) * bernulli_probability(2, 0, 7/10))

0.20140495867768599

c.

Ответ можно найти, посчитав вероятность, что не будет вытащего ни одного белого шара, и затем вычтем получившееся значение из единицы, тем самым определив вероятность всех остальных исходов

$P(A) \approx 99\%$

In [76]:
1 - bernulli_probability(2, 0, 9/11) * bernulli_probability(2, 0, 7/10)

0.9970247933884298